assert/equal.go

```diff
@@ -1,85 +1,135 @@
 package assert
 
 import (
-	"fmt"
 	"math"
 	"testing"
 )
 
 var ε = math.Nextafter(1, 2) - 1
 
+// EqualBools asserts two booleans for equality.
+// If they are not the equal then the failure function is called.
+func (assert *Assert) EqualBools(want bool, got bool, details ...interface{}) {
+	assert.t.Helper()
+	if want != got {
+		assert.fail("want[%t] but got[%t].%s", want, got, errordetails(details...))
+	}
+}
+
+// IsTrue asserts that b is true.
+// If it's not then the failure function is called with details.
+func (assert *Assert) IsTrue(b bool, details ...interface{}) {
+	assert.t.Helper()
+	assert.EqualBools(true, b, details...)
+}
+
+// IsFalse asserts that b is false.
+// If it's not then the failure function is called with details.
+func (assert *Assert) IsFalse(b bool, details ...interface{}) {
+	assert.EqualBools(false, b, details...)
+}
+
+// IsTrue asserts that b is true.
+// If it's not then Fatal() is called with details.
+func IsTrue(t *testing.T, cond bool, details ...interface{}) {
+	assert := New(t, Fatal)
+	assert.IsTrue(cond, details...)
+}
+
 // EqualStrings compares the two strings for equality.
-// If they are not equal t.Fatal is called using the details parameter.
-// The details parameter can be a single string of a format string + parameters.
+// If they are not equal then the failure function is called with details.
+func (assert *Assert) EqualStrings(want string, got string, details ...interface{}) {
+	assert.t.Helper()
+	if want != got {
+		detail := errordetails(details...)
+		assert.fail("wanted[%s] but got[%s].%s", want, got, detail)
+	}
+}
+
+// EqualStrings compares the two strings for equality.
+// If they are not equal then the Fatal() function is called with details.
 func EqualStrings(t *testing.T, want string, got string, details ...interface{}) {
 	t.Helper()
+	assert := New(t, Fatal)
+	assert.EqualStrings(want, got, details...)
+}
+
+// EqualInts compares the two ints for equality.
+// If they are not equal then the failure function is called with details.
+func (assert *Assert) EqualInts(want int, got int, details ...interface{}) {
+	assert.t.Helper()
 	if want != got {
 		detail := errordetails(details...)
-		t.Fatalf("wanted[%s] but got[%s].%s", want, got, detail)
+		assert.fail("wanted[%d] but got[%d].%s", want, got, detail)
 	}
 }
 
 // EqualInts compares the two ints for equality.
-// If they are not equal t.Fatal is called using the details parameter.
-// The details parameter can be a single string of a format string + parameters.
+// If they are not equal then the Fatal() function is called with details.
 func EqualInts(t *testing.T, want int, got int, details ...interface{}) {
```

Comment:
I'd like to change this to int64 comparison, what do you think?

Reply:
hmm won't that break a lot of code that uses just int and will have to cast it ? Usually I just use int's on code and most code I see also just uses int (the builtin `len` returns an int for example), so that could create extra work casting around if the default assert function is int64. I would be OK with providing another function for int64s, because sometimes we need those, but I think int's are more common :thinking: